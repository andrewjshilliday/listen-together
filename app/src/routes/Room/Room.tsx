import React, { useEffect } from 'react';
import { useRoom } from '../../components/providers';
import { QueueMediaItemView } from '../../components/common';
import styled from 'styled-components';

const Room: React.FC = (props: any) => {
  const roomProvider = useRoom();

  useEffect(() => {
    if (!roomProvider.roomId) {
      /* props.history.push('/'); */
    }
  }, [roomProvider.roomId]);

  useEffect(() => {
    console.log(roomProvider.playlist);
  }, [roomProvider.playlist]);
  
  return (
    <RoomContainer>
      <h1>Room {props.match.params.id} User {roomProvider.username}</h1>
      {roomProvider.playlist.map((item: MusicKit.MediaItem, index: number) => (
        <QueueMediaItemView key={item.id} item={item} index={index}></QueueMediaItemView>
      ))}
    </RoomContainer>
  );
}

export default Room;


const RoomContainer = styled.div``;
