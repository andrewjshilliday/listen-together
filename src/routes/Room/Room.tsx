import React, { useEffect } from 'react';
import { useRoom } from '../../components/providers';
import { QueueMediaItemView } from '../../components/common';
import './Room.scss';

const Room: React.FC = (props: any) => {
  const roomProvider = useRoom();

  useEffect(() => {
    console.log(roomProvider.playlist);
  }, [roomProvider.playlist]);
  
  console.log(roomProvider.playlist);
  return (
    <div className="room view">
      <h1>Room {props.match.params.id}</h1>
      {roomProvider.playlist.map((item: any, index: number) => (
        <QueueMediaItemView key={item.id} item={item} index={index}></QueueMediaItemView>
      ))}
    </div>
  );
}

export default Room;
