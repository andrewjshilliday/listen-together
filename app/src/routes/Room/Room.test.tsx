import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import renderer, { act } from 'react-test-renderer';
import Room from './Room';

jest.mock('../../components/providers/RoomProvider', () => require('../../__mocks__/providers/RoomProvider.mock').default);
jest.mock('../../components/providers/ChatProvider', () => require('../../__mocks__/providers/ChatProvider.mock').default);

jest.useFakeTimers();

beforeAll(() => {
  jest.resetModules();
})

it('renders correctly', async () => {
  const tree = renderer.create(
    <MemoryRouter initialEntries={[ '/room/TEST1' ]}>
      <Route path="/room/:id" component={Room} />
    </MemoryRouter>);

  jest.runTimersToTime(1);

  await act(async () => {
    await Promise.resolve();
  });

  expect(tree.toJSON()).toMatchSnapshot();
});
