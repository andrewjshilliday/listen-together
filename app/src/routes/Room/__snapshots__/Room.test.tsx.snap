// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h1
    className="sc-fzoiQi jkvegY"
  >
    Room 
    TEST1
     User 
    testuser
  </h1>,
  <div
    className="sc-fzqARJ blacQX"
  >
    <div
      className="sc-fzqNqU dugJCx"
    >
      <div
        className="sc-fzoyTs crHzLr"
        style={
          Object {
            "height": "100%",
            "overflow": "hidden",
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "bottom": 0,
              "left": 0,
              "marginBottom": 0,
              "marginRight": 0,
              "overflow": "scroll",
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <div
          style={
            Object {
              "borderRadius": 3,
              "bottom": 2,
              "display": "none",
              "height": 6,
              "left": 2,
              "position": "absolute",
              "right": 2,
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,.2)",
                "borderRadius": "inherit",
                "cursor": "pointer",
                "display": "block",
                "height": "100%",
                "position": "relative",
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "borderRadius": 3,
              "bottom": 2,
              "display": "none",
              "position": "absolute",
              "right": 2,
              "top": 2,
              "width": 6,
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,.2)",
                "borderRadius": "inherit",
                "cursor": "pointer",
                "display": "block",
                "position": "relative",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="sc-fzoXWK bsWqRD"
    >
      <div
        className="sc-fzokOt fwxvmr"
      >
        <div
          className="sc-fznxsB iCuWIt"
        >
          <span>
            Other people in this room
          </span>
          <div
            className="sc-fznJRM focDtY"
          >
            <span
              className="ant-avatar sc-fznWqX djDXSM ant-avatar-lg ant-avatar-circle"
              color="#cfcfcf"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={Object {}}
            >
              <span
                className="ant-avatar-string"
                style={
                  Object {
                    "opacity": 0,
                  }
                }
              >
                user0
              </span>
            </span>
            <span
              className="ant-avatar sc-fznWqX djDXSM ant-avatar-lg ant-avatar-circle"
              color="#cfcfcf"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={Object {}}
            >
              <span
                className="ant-avatar-string"
                style={
                  Object {
                    "opacity": 0,
                  }
                }
              >
                user1
              </span>
            </span>
          </div>
        </div>
        <div
          className="sc-fzqBZW hkmXeO"
        >
          <div
            className="sc-fzqNJr linzXx"
            style={
              Object {
                "height": "100%",
                "overflow": "hidden",
                "position": "relative",
                "width": "100%",
              }
            }
          >
            <div
              style={
                Object {
                  "WebkitOverflowScrolling": "touch",
                  "bottom": 0,
                  "left": 0,
                  "marginBottom": 0,
                  "marginRight": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <div
                className="sc-fzoyAV gGqGsQ"
              >
                <div
                  className="sc-fzoLag iwxmh"
                >
                  user0
                </div>
                <div
                  className="sc-fzoXzr KwfOU"
                />
              </div>
              <div
                className="sc-fzoyAV gGqGsQ"
              >
                <div
                  className="sc-fzoLag iwxmh"
                >
                  user1
                </div>
                <div
                  className="sc-fzoXzr KwfOU"
                />
              </div>
            </div>
            <div
              style={
                Object {
                  "borderRadius": 3,
                  "bottom": 2,
                  "display": "none",
                  "height": 6,
                  "left": 2,
                  "position": "absolute",
                  "right": 2,
                }
              }
            >
              <div
                style={
                  Object {
                    "backgroundColor": "rgba(0,0,0,.2)",
                    "borderRadius": "inherit",
                    "cursor": "pointer",
                    "display": "block",
                    "height": "100%",
                    "position": "relative",
                  }
                }
              />
            </div>
            <div
              style={
                Object {
                  "borderRadius": 3,
                  "bottom": 2,
                  "display": "none",
                  "position": "absolute",
                  "right": 2,
                  "top": 2,
                  "width": 6,
                }
              }
            >
              <div
                style={
                  Object {
                    "backgroundColor": "rgba(0,0,0,.2)",
                    "borderRadius": "inherit",
                    "cursor": "pointer",
                    "display": "block",
                    "position": "relative",
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="sc-fzpjYC hKVTug"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
            onKeyPress={[Function]}
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
            >
              Type a message
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
